manual/library/builtin.etex

```diff
@@ -190,7 +192,9 @@ exception Failure of string
 \index{Failure@\verb`Failure`}
 \begin{ocamldocdescription}
    Exception raised by library functions to signal that they are
-   undefined on the given arguments.
+   undefined on the given arguments.  The string gives some information
+   to the programmer; it is \emph{bad practice} to pattern match
+   on it, \verb`Failure _` should be used.
```

Comment:
You could be a bit more explicit "pattern match on *the string literal*", and explicit (and in the example below as well) that the string literal may change in future versions to clarify error messages.